Source/WebCore/platform/cocoa/WebAVPlayerLayer.mm

```diff
@@ -120,6 +148,20 @@ - (CALayer*)videoSublayer
     return _videoSublayer.get();
 }
 
+- (CGSize)videoDimensions
+{
+    return _videoDimensions;
+}
+
+- (void)setVideoDimensions:(CGSize)videoDimensions
+{
+    if (CGSizeEqualToSize(_videoDimensions, videoDimensions))
+        return;
+
+    _videoDimensions = videoDimensions;
+    [self setNeedsDisplay];
```

Comment:
This might trigger backing store allocation. Do you want `setNeedsLayout` instead?

Reply:
Whoops, I do mean `setNeedsLayout`.